solutions/logging/core-env/project-iam.yaml

```diff
@@ -0,0 +1,126 @@
+spec:
+  resourceRef:
+    kind: Project
+    name: logging-prj-id # kpt-set: ${logging-prj-id}
+    namespace: projects
+  # AC-3(7) ACCESS ENFORCEMENT | ROLE-BASED ACCESS CONTROL
+  bindings:
+    - role: roles/logging.bucketWriter
+      members:
+        - memberFrom:
+            logSinkRef:
+              name: logging-prj-id-security-sink # kpt-set: ${logging-prj-id}-security-sink
+              namespace: logging
+---
+# Logs Bucket writer IAM permissions for the platform and component log sinks
+# Binds the generated writer identity from the LoggingLogSink to the logging project
+# AC-3(7) - Write access to the logging buckets is limited by IAM to just the identities of the log sinks configured to send logs to the buckets (set at the logging project level)
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPartialPolicy
+metadata:
+  name: platform-and-component-services-log-bucket-writer-permissions
+  namespace: projects
+  annotations:
+    config.kubernetes.io/depends-on: resourcemanager.cnrm.cloud.google.com/namespaces/projects/Project/${logging-prj-id} # kpt-set: resourcemanager.cnrm.cloud.google.com/namespaces/projects/Project/${logging-prj-id}
+spec:
+  resourceRef:
+    kind: Project
+    name: logging-prj-id # kpt-set: ${logging-prj-id}
+    namespace: projects
+  # AC-3(7) ACCESS ENFORCEMENT | ROLE-BASED ACCESS CONTROL
+  bindings:
+    - role: roles/logging.bucketWriter
+      members:
+        - memberFrom:
+            logSinkRef:
+              name: platform-and-component-services-log-sink
+              namespace: logging
+---
+# Logs Bucket writer IAM permissions for the platform and component services log sinks
+# Binds the generated writer identity from the LoggingLogSink to the logging project
+# AC-3(7) - Write access to the logging buckets is limited by IAM to just the identities of the log sinks configured to send logs to the buckets (set at the logging project level)
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPartialPolicy
+metadata:
+  name: platform-and-component-services-infra-log-bucket-writer-permissions
+  namespace: projects
+  annotations:
+    config.kubernetes.io/depends-on: resourcemanager.cnrm.cloud.google.com/namespaces/projects/Project/${logging-prj-id} # kpt-set: resourcemanager.cnrm.cloud.google.com/namespaces/projects/Project/${logging-prj-id}
+spec:
+  resourceRef:
+    kind: Project
+    name: logging-prj-id # kpt-set: ${logging-prj-id}
+    namespace: projects
+  # AC-3(7) ACCESS ENFORCEMENT | ROLE-BASED ACCESS CONTROL
+  bindings:
+    - role: roles/logging.bucketWriter
+      members:
+        - memberFrom:
+            logSinkRef:
+              name: platform-and-component-services-infra-log-sink
+              namespace: logging
+---
+# Logs Bucket writer IAM permissions for the platform and component log sink
+# Binds the generated writer identity from the LoggingLogSink to the logging project
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPartialPolicy
```

Comment:
I believe all these distinct IAMPartialPolicy can be combined in a single resource as they are all referencing the same resource. i.e. logging-project-id. please investigate

Reply:
Resolved: Decision is to leave as is, and make part of IAM policy packages (authoritative) in future.